app/grandchallenge/webhooks/tasks.py

```diff
@@ -0,0 +1,61 @@
+import json
+from datetime import datetime
+
+import jwt
+import requests
+from celery import shared_task
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from django.apps import apps
+from django.conf import settings
+from django.core import files
+from django.core.files.temp import NamedTemporaryFile
+
+
+@shared_task()
+def get_tarball(*, pk):
+    GitHubWebhookMessage = apps.get_model(  # noqa: N806
+        app_label="webhooks", model_name="GitHubWebhookMessage"
+    )
+    ghwm = GitHubWebhookMessage.objects.get(pk=pk)
+    payload = ghwm.payload
+    installation_id = payload["installation"]["id"]
+    with open(settings.GITHUB_PRIVATE_KEY_PATH, "rb") as key_file:
+        private_key = serialization.load_pem_private_key(
+            key_file.read(), password=None, backend=default_backend()
+        )
+    now = datetime.now()
+    msg = {
+        "iat": int(now.timestamp()) - 60,
+        "exp": int(now.timestamp()) + 60 * 10,
+        "iss": settings.GITHUB_APP_ID,
+    }
+    token = jwt.encode(msg, private_key, algorithm="RS256")
```

Comment:
Has this been double checked? JWTs have a bad rep for being hard to get right, I think with the algorithm explicitally set we're ok but I'm unsure.

Reply:
It took me forever to get right, so that rep is well earned I'd say. This seems to work properly though.